substrate/client/authority-discovery/src/worker.rs

```diff
@@ -946,6 +990,21 @@ where
 	}
 }
 
+/// Removes the `/p2p/..` from the address if it is present.
+fn address_without_p2p(mut address: Multiaddr, local_peer_id: PeerId) -> Multiaddr {
+	if let Some(multiaddr::Protocol::P2p(peer_id)) = address.iter().last() {
+		if peer_id != *local_peer_id.as_ref() {
+			error!(
```

Comment:
Having some sort of "context" here to know from where the `address` is coming from would be nice. Also for things like `external_addresses`, we should not print anything at all? I mean these addresses are coming from other nodes, but we probably ensure before that these contain our peer id?

Reply:
Yep that makes sense! Have added an `AddressType` enum for better logs 🙏 

While at it, have kept the error for all possible address types (including external addresses), since we don't do any validation in the lower components (libp2p / litep2p).

I'll come with a followup, there are a few things I'd like to check:
- move the external address error directly into libp2p / litep2p discovery mechanism
- don't emit events for `ExternalAddressConfirmed` for addresses that have a different peer ID
- libp2p looks like it adds `/p2p/` regardless of the address already containing it

https://github.com/paritytech/polkadot-sdk/blob/9fed62ea4703c2a77e06df801b9820a353b77972/substrate/client/network/src/discovery.rs#L752